.github/workflows/ci.yml

```diff
@@ -42,20 +84,34 @@ jobs:
   sw-snitch-cluster-vlt:
     name: Simulate SW on Snitch Cluster w/ Verilator
     runs-on: ubuntu-22.04
+    needs: build-docker
+    if: >
+      github.event_name != 'pull_request' ||
+      github.event.pull_request.head.repo.full_name != github.repository
     container:
-      image: ghcr.io/pulp-platform/snitch_cluster:main
+      image: ghcr.io/pulp-platform/snitch_cluster:${{ github.ref_name }}
     steps:
       - uses: actions/checkout@v2
         with:
           submodules: 'recursive'
-      - name: Install local Python packages
-        run: pip install .
       - name: Build Software
         working-directory: target/snitch_cluster
         run: |
-          bender vendor init
           make CFG_OVERRIDE=cfg/github-ci.hjson sw
+      - name: Generate the RTL
+        working-directory: target/snitch_cluster
+        run: |
+          make CFG_OVERRIDE=cfg/github-ci.hjson rtl
+      - name: Set up cache for Verilator build
+        id: verilator-cache
+        uses: actions/cache@v3
+        with:
+          path: target/snitch_cluster/bin
+          key: verilator-${{ hashFiles('Bender.lock', 'hw/**/*.sv', 'target/**/*.sv') }}
```

Comment:
Yes, I know. This was actually  my first approach with a `vlt-flist` make target. But the `hashFiles` function is a bit annyoing, since it requires comma-separated files, which I didn't have the energy anymore to debug😄 

If you want to pick it up, this is the commit https://github.com/pulp-platform/snitch_cluster/pull/192/commits/3dfdbfc268a143e2807037998036daf9d7d4155c where I reverted to the simpler hashing of files